src/main/java/org/tndata/android/compass/activity/GoalDetailsActivity.java

```diff
@@ -0,0 +1,233 @@
+    // A user may tap an action or behavior to see more info
+    private Behavior mSelectedBehavior;
+    private Action mSelectedAction;
+
+    private GoalDetailsFragment mGoalDetailsFragment = null;
+    private LearnMoreFragment mLearnMoreFragment = null;
+    private ArrayList<Fragment> mFragmentStack = new ArrayList<Fragment>();
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_base);
+
+        mCategory = (Category) getIntent().getSerializableExtra("category");
+        mGoal = (Goal) getIntent().getSerializableExtra("goal");
+
+        mToolbar = (Toolbar) findViewById(R.id.tool_bar);
+        mToolbar.setTitle(mGoal.getTitle());
+        mToolbar.getBackground().setAlpha(255);
+        mToolbar.setNavigationIcon(R.drawable.ic_arrow_back_white);
+        setSupportActionBar(mToolbar);
+        getSupportActionBar().setDisplayShowHomeEnabled(true);
+
+        if (mCategory != null && !mCategory.getColor().isEmpty()) {
+            mToolbar.setBackgroundColor(Color.parseColor(mCategory.getColor()));
+        }
+        swapFragments(GOALDETAIL, true);
+    }
+
+    @Override
+    public boolean onKeyDown(int keyCode, KeyEvent event) {
+        if ((keyCode == KeyEvent.KEYCODE_BACK)) { // Back key pressed
+            handleBackStack();
+            return true;
+        }
+        return super.onKeyDown(keyCode, event);
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        switch (item.getItemId()) {
+            case android.R.id.home:
+                handleBackStack();
+                return true;
+        }
+        return super.onOptionsItemSelected(item);
+    }
+
+    @Override
+    public void learnMoreGoal(Goal goal) {
+        mGoal = goal;
+        swapFragments(LEARN_MORE_GOAL, true);
+    }
+
+    @Override
+    public void learnMoreBehavior(Behavior behavior) {
+        mSelectedBehavior = behavior;
+        swapFragments(LEARN_MORE_BEHAVIOR, true);
+    }
+
+    @Override
+    public void learnMoreAction(Action action) {
+        mSelectedAction = action;
+        swapFragments(LEARN_MORE_ACTION, true);
+    }
+
+    @Override
+    public void chooseBehaviors(Goal goal) {
+        Intent intent = new Intent(getApplicationContext(), GoalTryActivity.class);
+        intent.putExtra("goal", goal);
+        intent.putExtra("category", mCategory);
+        startActivityForResult(intent, Constants.CHOOSE_BEHAVIORS_REQUEST_CODE);
+    }
+
+    @Override
+    public void deleteGoal(Goal goal) {
+        mFragmentStack.clear();
+
+        // Delete the goal.
+        ArrayList<String> goals = new ArrayList<String>();
+        goals.add(String.valueOf(goal.getMappingId()));
+        new DeleteGoalTask(this, this, goals).executeOnExecutor(AsyncTask.THREAD_POOL_EXECUTOR);
+
+        setResult(Constants.GOALS_CHANGED_RESULT_CODE);
+    }
+
+    @Override
+    public void goalsDeleted() {
```

Comment:
I would do it before sendBroadcast(), but also, since you are calling setResult(Constants.GOALS_CHANGED_RESULT_CODE); I would make sure of two things: 1 - use startActivityForResult() to start this activity, and 2 - in the activity that started this, use onActivtyResult() to check for this result code, and make sure the UI is updated properly.